### ISSUE TITLE
Aria label being forced always

### ISSUE THREAD

**Issue by zonakusu** [ionic-team/ionicons#710]
Hi everyone :)

I have a small issue regarding accessibility using `ion-button` and `ion-icon`. While using this in combination my screen reader always reads both, even if I'm using aria-hidden on an element.

For me, the most semantic would be to see the `ion-button` to be the element which is read by the screen reader, while the `ion-icon` is ignored.

Problem though, is that the `aria-label` is being forced, regardless if I set `ariaLabel` to false, null or empty, or when using `aria-hidden`. Cause for this behaviour if the [following block of code](https://github.com/ionic-team/ionicons/blob/master/src/components/icon/icon.tsx#L127).

Of course I love the default `aria-label` in terms of accessibility, however I would like to control it just a little bit better :)

I would like to create a PR enabling to either not set a default `aria-label` or disabling `aria-label` by setting the `aria-hidden`. My suggestion would be using `[ariaLabel]="null"` since it would not introduce a new binding. 

What do you guys think? Agree with my thoughts? Please spare no comments :)

**Comment by peterpeterparker** [comment]
I have provided a PR [#869](https://github.com/ionic-team/ionicons/pull/869)

**Comment by peterpeterparker** [comment]
PR has been merged, I think this issue can now  be closed. No label will be generated automatically if `aria-hidden` will be used, as for example:

```
<ion-icon name="cellular" aria-hidden="true"></ion-icon>
```

**Comment by chriszuercher** [comment]
I tried setting aria-hidden="true" on all ion-icon but the SVG label is still generated and shown using version 5.5.1. Also adding an aria-label to the icons is not working. Is there something else to do? The behaviour changed between 5.5 and 5.1.0 (in 5.1 there was no text generated). But I need one icon of 5.5 so downgrad is not really an option.

We are using ion-icons in an asp.net core application with the following script include:
`        <script type="module" src="~/lib/ionicons/ionicons/ionicons.esm.js"></script>
        <script nomodule src="~/lib/ionicons/ionicons/ionicons.min.js"></script>`

**Comment by peterpeterparker** [comment]
What do you mean with "the SVG label" ?

Your example is correct, with `aria-hidden` set to `true` no `aria-label` is automatically generated.

From the `index.html` example of the project:

```
<ion-icon name="cellular"></ion-icon>

output -> <ion-icon name="cellular" role="img" class="ios hydrated" aria-label="cellular"></ion-icon>

<ion-icon name="cellular" aria-label="Mobile data"></ion-icon>

output -> <ion-icon name="cellular" aria-label="Mobile data" role="img" class="ios hydrated"></ion-icon>

<ion-icon name="cellular" aria-hidden="true"></ion-icon>

output -> <ion-icon name="cellular" aria-hidden="true" role="img" class="ios hydrated"></ion-icon>
```

all good to me.

**Comment by MicheleLucini-infobit** [comment]
aria-hidden="true" will prevent the aria-label of the <ion-icon /> to appear but the svg shadowroot in the tag <ion-icon /> still has the tag title that appear on hover

**Comment by MicheleLucini-infobit** [comment]
aria-hidden="true" will prevent the aria-label of the `<ion-icon />` to appear but the svg shadowroot in the tag `<ion-icon />` still has the tag title that appear on hover

**Comment by MicheleLucini-infobit** [comment]
![image](https://user-images.githubusercontent.com/73997474/118134139-df583900-b401-11eb-8243-2538cbc44321.png)

**Comment by MicheleLucini-infobit** [comment]
Current workaround 😖
```
ion-icon {
  pointer-events: none;
}
```

**Comment by peterpeterparker** [comment]
[MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-hidden_attribute)

> Adding aria-hidden="true" to an element removes that element and all of its children from the accessibility tree. 

As for example

```
<p aria-hidden="true" title="hello">
  Some things are better left unsaid.
</p>
```

will display "hello" on hover too.

**Comment by MicheleLucini-infobit** [comment]
Maybe i'm in the wrong thread, 
I don't want the  `<title>Ellipsis Vertical</title>` to show on hover, how can i prevent that?
![image](https://user-images.githubusercontent.com/73997474/118137859-f9941600-b405-11eb-9154-6f4d36ef880f.png)

**Comment by MicheleLucini-infobit** [comment]
I'm using ionicon 5.5.1 and my react code is the following `<ion-icon name={icon} class={`${btnIcon} mg-r-${iconTextMargin}`} aria-hidden="true" title="" />`

**Comment by peterpeterparker** [comment]
@MicheleLucini-infobit  so yes, I think too that your issue has not to do with OP's issue aka wrong thread.

Then you workaround is alright. Another one would be to download / adding the icon to your project and removing the `title` from the svg.

Might be worth to submit a dedicated PR or issue.

**Comment by brandyscarney** [comment]
Thanks for the issue, closing this to track in #838.
